git_best_practices.md

```diff
@@ -0,0 +1,25 @@
+# Git Best Practices
+
+---
+
+Here are a few easy ways to make sure your PR gets reviewed quickly, and your code gets merged as efficiently as possible.
+
+## Scope your branches!
+
+- To avoid name collsions please prefix your branch names with your name (e.g. use `sharif-my-cool-feature` not `my-cool-feature`)
```

Comment:
for naming we could use `/` to separate the dev and branch name eg `sharif/<branch_name>` too, then these branches will show up as collapsible by dev name